src/Codeception/Module/Db.php

```diff
@@ -232,28 +243,58 @@ private function disconnect()
         $this->driver = null;
     }
 
+    public function _beforeSuite($settings = [])
+    {
+        $this->connect();
+
+        // FIXME: This needs proper structure to avoid cooking more spaghetti code
+        // since it only concerns to the "legacy" dump loading procedure.
+        // The db drivers should be revised to see if they are coupled on
+        // having the $sql variable loaded at this point in time of the execution or not...
+        if (!$this->config['dumptool']
```

Comment:
I don't like spaghetti code as well, so please move dump loaders to corresponding classes